### ISSUE TITLE
Slice assignment of chararrays broken in Python 3 (Trac #1817)

### ISSUE THREAD

**Issue by numpy-gitbot** [numpy/numpy#2410]
_Original ticket http://projects.scipy.org/numpy/ticket/1817 on 2011-05-03 by trac user embray, assigned to unknown._

I'm not sure if this is by design or not, but it's certainly surprising:

```
#!python
>>> charar = np.chararray((3, 3), itemsize=5)
>>> charar[:] = 'abcde'
>>> charar
chararray([[b'\xa8:\x15~8', b'\x00\x00\x00\xa8:', b'\x15~8'],
       [b'\x00\xd0\xbc\x9b', b'\x00\x00\x00\x00\xb0', b'\xac\x9b'],
       [b'\x00\x003', '', b'\x10=\xd2']], 
      dtype='|S5')
```

Works as intended on Python 2.x.

**Comment by saimn** [comment]
Seems to be fixed, I guess by #5982

**Comment by mhvk** [comment]
Indeed, seems fixed, so closing! Thanks for noticing, @saimn
